### ISSUE TITLE
yr.no will discontinue XML API in the first quarter 2021

### ISSUE THREAD

**Issue by SirUli** [Martin-Gleiss/smartvisu#379]
While debugging an issue with the yr.no weather service, i stumbled across this text in their XML response:

```xml
<apiChanges>
<!-- NB!!! Since we have no other options to contact our users of forecast.xml, we have decided to use this method to get your attention to some major changes with our API -->
<!-- In the first quarter next year, forecast.xml and forecast_hour_by_hour.xml will be discontinued in favour of a newer JSON API. -->
<!-- Vi encourage you already now to switch to the new API. -->
<!-- There are a lot of changes, and you can read all the technical details in our new developer portal https://developer.yr.no -->
</apiChanges>
```

I would assume that it might be worth to switch in the long run ;)

Best Regards,
Uli

**Comment by wvhn** [comment]
Thanks for the hint. I've already seen this, too but did not find a proper documentation of the new json API on yr.no. Is there any chance you could contribute to the solution?  

Best regards,
Wolfram

**Comment by SirUli** [comment]
Hi @wvhn 

i guess the API is this one: https://api.met.no/weatherapi/locationforecast/2.0/documentation

Unfortunately there are a number of changes (e.g. it requires lat/long data instead of a city name) so this won't be backwards compatible.

I will take a look at it, but not yet sure if i will be using this type of forecast in the long run (since my FHEM adapter already provides the data in theory). So i cannot promise anything yet :)

Cheers,
Uli

**Comment by wvhn** [comment]
That's exactly the point. The documentation is from met.no and it can not be seen whether yr.no will continue to accept location names instead of lat/long.

Thx
Wolfram

**Comment by SirUli** [comment]
I guess i skipped a [part](https://hjelp.yr.no/hc/en-us/articles/206550539-Om-Yr).

> Yr is the joint online weather service from the Norwegian Meteorological Institute (met.no) and the Norwegian Broadcasting Corporation (NRK).

So "met" is indeed the new api endpoint. The link originally came from here: https://developer.yr.no/doc/GettingStarted/

Looking through the API docs, there is no possibility of setting the location only. --> I would rather add a weather service called "met" than updating the yr.no one.

**Comment by wvhn** [comment]
solved with PR #486 
New weather service "met.no" is replacement for "yr.no"
